File: OLSource1/v1/VS_officedev/codesnippet/CSharp/walkthrough--creating-a-template-by-using-content-controls_4.cs
                Word.Template template1 = this.AttachedTemplate as Word.Template;

                if (template1 != null)
                {
                    object description = null;
                    template1.BuildingBlockEntries.Add("Employee Table",
                        Word.WdBuildingBlockTypes.wdTypeCustom1, "Employee and Customer Information",
                        this.Tables[1].Range, ref description, Word.WdDocPartInsertOptions.wdInsertContent);
                    template1.BuildingBlockEntries.Add("Customer Table",
                        Word.WdBuildingBlockTypes.wdTypeCustom1, "Employee and Customer Information",
                        this.Tables[2].Range, ref description, Word.WdDocPartInsertOptions.wdInsertContent);
                }

File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/walkthrough--creating-a-template-by-using-content-controls_4.vb
            Dim template1 As Word.Template = TryCast(Me.AttachedTemplate, Word.Template)
            If template1 IsNot Nothing Then

                template1.BuildingBlockEntries.Add("Employee Table", _
                    Word.WdBuildingBlockTypes.wdTypeCustom1, "Employee and Customer Information", _
                    Me.Tables(1).Range, InsertOptions:=Word.WdDocPartInsertOptions.wdInsertContent)
                template1.BuildingBlockEntries.Add("Customer Table", _
                    Word.WdBuildingBlockTypes.wdTypeCustom1, "Employee and Customer Information", _
                    Me.Tables(2).Range, InsertOptions:=Word.WdDocPartInsertOptions.wdInsertContent)
            End If